rewrite-test/src/main/java/org/openrewrite/test/RewriteTest.java

```diff
@@ -157,13 +157,23 @@ default void rewriteRun(Consumer<RecipeSpec> spec, SourceSpec<?>... sourceSpecs)
                 .isNotNull();
 
         if (!(recipe instanceof AdHocRecipe) && !(recipe instanceof CompositeRecipe) &&
+            !(recipe.equals(Recipe.noop())) &&
             testClassSpec.serializationValidation &&
             testMethodSpec.serializationValidation) {
             RecipeSerializer recipeSerializer = new RecipeSerializer();
             assertThat(recipeSerializer.read(recipeSerializer.write(recipe)))
                     .as("Recipe must be serializable/deserializable")
                     .isEqualTo(recipe);
-            assertThatCode(() -> RecipeIntrospectionUtils.constructRecipe(recipe.getClass()))
+            assertThatCode(() -> {
+                Recipe r = RecipeIntrospectionUtils.constructRecipe(recipe.getClass());
+                // getRecipeList should not fail with default parameters from RecipeIntrospectionUtils.
+                r.getRecipeList();
+                // We add recipes to HashSet in some places, we need to validate that hashCode and equals does not fail.
+                //noinspection ResultOfMethodCallIgnored
+                r.hashCode();
+                //noinspection EqualsWithItself
+                assert r.equals(r);
```

Comment:
Maybe this won't hold up in test environments, but whenever I see an assert statement I tend to think they [might not be enabled at runtime](https://docs.oracle.com/javase/8/docs/technotes/guides/language/assert.html#enable-disable):
> By default, assertions are disabled at runtime.

Would it then not be better to not use assert here?